### ISSUE TITLE
Regreassion: RF v7.0a1 doesn't support using `staticmethod` or `classmethod` with `keyword` and `library`

### ISSUE THREAD

**Issue by richardstone** [robotframework/robotframework#4941]
I’m trying out the new RF 7.0a1 and it brought up a strange error. I have a couple keywords defined in separate python library and after the upgrade to the RF 7.0 alpha build I started to experience issues with the keywords in my file. Basically the library looks like this:
```
#!/usr/bin/env python3

import subprocess

from robot.api.deco import library, keyword

@library(scope='GLOBAL', auto_keywords=False)
class UtilsPythonLib:

    @staticmethod
    @keyword(name="Get Hostname")
    def get_hostname():
        return subprocess.run('hostname',
                              stdout=subprocess.PIPE).stdout.decode('utf-8').replace("\n", "")

    @staticmethod
    @keyword(name='Get Cluster Type For Hostname')
    def get_cluster_type_for_hostname(hostname):
        for cluster_type in HOSTNAME_CLUSTER:
            for cluster_name in HOSTNAME_CLUSTER[cluster_type]:
                if hostname in cluster_name:
                    return cluster_type
        return None
```
I have this imported and used in the resource file:
```
*** Settings ***
Documentation       Collection of system information related keywords

Library             ../python/UtilsPythonLib.py

Skip tests if cluster type is
    [Arguments]    ${cluster_to_skip}
    ${host_name} =    Get Hostname
    ${cluster_type} =    Get Cluster Type For Hostname    ${host_name}
    Skip If    "${cluster_type}" == "${cluster_to_skip}"
```

This worked on RF v6.1.1 but I see issues with RF v7.0a1.
I have tried to swap the order of the decorators, and that cured my problem.

**Comment by pekkaklarck** [comment]
This regression is caused by the fix for #4915 committed in 523d9dba31378dc9199dc2aa9b8bcdaddc513955. As part of the fix I changed the code that checks is the `keyword` decorator used to get the keyword method with `inspect.getattr_static` (it avoids calling properties and is simpler than the old code). When using `staticmethod` or `classmethod`, we nowadays get those instances instead of the methods they wrap. The fix is to get the wrapped method in these special cases from from the wrapper explicitly.

While investigating this, I noticed we didn't have any tests for using `staticmethod` with keywords methods in the first place. I'll add simple tests for it along with tests for validating that `staticmethod` and `classmethod` work with `keyword`.

**Comment by pekkaklarck** [comment]
I won't add this issue to RF 7.0 milestone because this isn't a bug in earlier versions that should be mentioned in the release notes. In practice we should actually reopen #4915 because it introduced this issue, but because I have a fix for the issue already done locally that doesn't make much sense either.
